ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+                spacer + comments[i] + '\n'
+
+        return ep_string
+
+    @property
+    def name(self):
+        """Get or set the name."""
+        return self._name
+
+    @name.setter
+    def name(self, data):
+        assert isinstance(data, str), 'Design day name must be' \
+            ' text. Got {}'.format(type(data))
+        self._name = data
+
+    @property
+    def day_type(self):
+        """Get or set the type of design day."""
+        return self._day_type
+
+    @day_type.setter
+    def day_type(self, data):
+        assert data in day_types, 'day_type {} is not' \
+            ' recognized'.format(str(data))
+        self._day_type = data
+
+    @property
+    def location(self):
+        """Get or set the location."""
+        return self._location
+
+    @location.setter
+    def location(self, data):
+        assert hasattr(data, 'isLocation'), 'Expected' \
+            ' isLocation type. Got {}'.format(str(data))
+        self._location = data
+
+    @property
+    def dry_bulb_condition(self):
+        """Get or set the dry bulb conditions."""
+        return self._dry_bulb_condition
+
+    @dry_bulb_condition.setter
+    def dry_bulb_condition(self, data):
+        assert hasattr(data, 'isDryBulbCondition'), 'Expected' \
+            ' DryBulbCondition type. Got {}'.format(str(data))
+        self._dry_bulb_condition = data
+
+    @property
+    def humidity_condition(self):
+        """Get or set the humidity conditions."""
+        return self._humidity_condition
+
+    @humidity_condition.setter
+    def humidity_condition(self, data):
+        assert hasattr(data, 'isHumidityCondition'), 'Expected' \
+            ' HumidityCondition type. Got {}'.format(str(data))
+        self._humidity_condition = data
+
+    @property
+    def wind_condition(self):
+        """Get or set the wind conditions."""
+        return self._wind_condition
+
+    @wind_condition.setter
+    def wind_condition(self, data):
+        assert hasattr(data, 'isWindCondition'), 'Expected' \
+            ' WindCondition type. Got {}'.format(str(data))
+        self._wind_condition = data
+
+    @property
+    def sky_condition(self):
+        """Get or set the sky conditions."""
+        return self._sky_condition
+
+    @sky_condition.setter
+    def sky_condition(self, data):
+        assert hasattr(data, 'isSkyCondition'), 'Expected' \
+            ' SkyCondition type. Got {}'.format(str(data))
+        self._sky_condition = data
+
+    @property
+    def analysis_period(self):
+        """The analysisperiod of the design day."""
+        return AnalysisPeriod(
+            self.sky_condition.month,
+            self.sky_condition.day_of_month,
+            0,
+            self.sky_condition.month,
+            self.sky_condition.day_of_month,
+            23)
+
+    @property
+    def hourly_datetimes(self):
+        return self._sky_condition._get_datetimes()
+
+    @property
+    def hourly_dry_bulb_data(self):
```

Comment:
Do we need data at the end of the name for all these methods?

Reply:
No. It has been removed.